### PR TITLE
Make independent

### PR DESCRIPTION
**Pull request by levitte**
I thought you guys might be interested in changes to make this independent from the OpenSSL source tree (except for the mkerr.pl part, but that's not critical for building).

### CODE REVIEW

File: gost_ameth.c [hunk 1]
```diff
@@ -57,10 +57,15 @@ static int pkey_bits_gost(const EVP_PKEY *pk)
 
     switch (EVP_PKEY_base_id(pk)) {
     case NID_id_GostR3410_2001:
+        return 256;
+#if defined(NID_id_GostR3411_2012_256)
     case NID_id_GostR3410_2012_256:
         return 256;
+#endif
+#if defined(NID_id_GostR3411_2012_256)
```
Comment (mkosmach) [inline gost_ameth.c]: Do you mean
#if defined(NID_id_GostR3411_2012_512)
in line 65?

Comment (beldmit) [inline gost_ameth.c]: They were defined at the same time in one commit. So if your patch is applied nothing but clarity will change.

Comment (mkosmach) [inline gost_ameth.c]: It's not my patch.
I mean that in line 65 NID_id_GostR3411_2012_512 should be checked instead of NID_id_GostR3411_2012_256.

Comment (beldmit) [inline gost_ameth.c]: In real life there is no difference.

**Comment by mouse07410** [comment]
@levitte it would be great if you could rebase your PR... One good thing about this engine is it's complete coverage of the new Russian standard. One bad thing about it is that it doesn't even compile as-is. I hope that with your PR one might be able to get it working with OpenSSL-1.1 at least (the code comments claim it can be compiled with any OpenSSL after 1.0.0, but if you believe that claim - I have a nice bridge for sale :).

**Comment by levitte** [comment]
Sure, I'm on it now.

**Comment by levitte** [comment]
Oh, I see you guys went with cmake!  Ok then, I'll ditch the auto-* stuff then.

**Comment by levitte** [comment]
Actually, this PR needs a complete makeover, to the point of starting over entirely.  gost-engine has obviously been adapted for 1.1.0 and on only, am I correct?

**Comment by beldmit** [comment]
Yes, the master branch of the gost-engine should be compatible with 1.1.* branches.
Richard, thank you!

**Comment by levitte** [comment]
Closed in favor of #26 and #27

**Comment by mouse07410** [comment]
> Oh, I see you guys went with cmake! Ok then, I'll ditch the auto-* stuff then.

@levitte please do **not** ditch the auto- stuff! It is useful, and not everybody is happy with CMake.

Also, maybe you could look back at https://github.com/engine-corner/Lesson-2-A-digest/pull/2 and hopefully merge it. ;-)

**Pull request closed without merging**